lib/utils.js

```diff
@@ -53,3 +53,8 @@ export async function loadGetInitialProps (Component, ctx) {
   }
   return props
 }
+
+export function getLocationOrigin () {
+  const { protocol, hostname, port } = window.location
+  return `${protocol}//${hostname}${port ? ':' + port : ''}`
```

Comment:
It's valid, works fine: https://github.com/relatenow/relate/blob/b0181ac9d80e2fee3ab2387e25342c0837bf789f/util/baseUrl.js#L4

```js
`${protocol}//${hostname}${port ? `:${port}` : ''}`
```